app/src/main/java/com/mathewsachin/fategrandautomata/imaging/DroidCvPattern.kt

```diff
@@ -75,9 +92,23 @@ class DroidCvPattern(private var Mat: Mat? = Mat(), private val OwnsMat: Boolean
             val result = DisposableMat()
 
             if (Template.width <= width && Template.height <= height) {
-                Imgproc.matchTemplate(Mat, Template.Mat, result.Mat, Imgproc.TM_CCOEFF_NORMED)
-            }
-            else logd("Skipped matching $Template: Region out of bounds")
+                alpha.let { mask ->
```

Comment:
Doesn't this let only create an alias for the `alpha` variable?

Why not use alpha directly?

Reply:
Because, it is a global variable, once you've checked for null, you can't be sure in the next line that it won't be set null again.

Reply:
You're not checking for null. Normally you would use `let` this way:
`nullable?.let {}`

Reply:
I'm checking for `null` inside `let`. Maybe try compiling yourself to get the intuition.

Reply:
Ok. my fault. OpenCV doesn't care about null 🤦

Reply:
removed the `let`.